modules/Bio/EnsEMBL/Compara/PipeConfig/CreateSpeciesTree_conf.pm

```diff
@@ -235,6 +235,7 @@ sub pipeline_analyses {
           -parameters => {
               'taxonomic_ranks' => $self->o('taxonomic_ranks'),
               'custom_groups'   => $self->o('custom_groups'  ),
+              'genome_dumps_dir' => $self->o('genome_dumps_dir'),
```

Comment:
It's unclear to me where `genome_dumps_dir` gets set in the first place, it isn't in the `default_options` - possibly an eHive default somewhere. I simply copied the way it was already being passed to `check_sketches` in the PipeConfig.

Reply:
[ENV.pm](https://github.com/Ensembl/ensembl-compara/blob/master/modules/Bio/EnsEMBL/Compara/PipeConfig/ENV.pm#L83
)
[ComparaGeneric_conf.pm](https://github.com/Ensembl/ensembl-compara/blob/master/modules/Bio/EnsEMBL/Compara/PipeConfig/ComparaGeneric_conf.pm#L45)
[CreateSpeciesTree_conf.pm](https://github.com/Ensembl/ensembl-compara/blob/master/modules/Bio/EnsEMBL/Compara/PipeConfig/CreateSpeciesTree_conf.pm#L47)
Here is a breadcrumb trail :slightly_smiling_face:

Reply:
It is fine to do it the way you have done it, but I would be tempted to move all of those specific parameter calls - because they are needed in most analyses, to `pipeline_wide_parameters` - including the one in `check_sketches` - just to reduce the code repetition. And in `pipeline_wide_parameters` it is nice to have a quick look in the guihive too.

Reply:
I think if this is a stylistic choice in that manner I'm opposed to it because it actually makes it _less_ clear which parameters a particular step requires/uses at a glance.

Reply:
Up to you, but this wouldn't make a difference in the case of parameters that are passed from the runnables prior to the analysis - so for the sake of consistency and wholesome "knowing what parameters a particular step requires" - you will find that it is more than `genome_dumps_dir`. We generally pass them in individually in this way if we want to override the `default` or `pipeline_wide_parameters` - which we do not in this case.